src/features/sign-in/components/SgidCallback.tsx

```diff
@@ -18,6 +21,8 @@ export const SgidCallback = () => {
     },
     {
       onSuccess: async ({ redirectUrl }) => {
+        setHasLoginStateFlag()
+        await utils.me.get.invalidate()
```

Comment:
the problemmmmmm with this, is that even on an error, `onSuccess` happens haha, because we want to return a `redirectUrl`. We may need to add another flag to denote "true" success

Reply:
alternatively (and probably the more correct one), is to throw an error on the backend and redirect on the frontend to the correct route instead:

for example,
```ts
        throw new TRPCError({
          code: 'BAD_REQUEST',
          message: 'SGID is not enabled',
        })
```

then on the frontend:

```tsx
trpc.auth.sgid.callback.useSuspenseQuery(
    {
      code: String(code),
      state: String(state),
    },
    {
      onSuccess: async ({ redirectUrl }) => {
        // as before
      },
      onError: async (error) => {
        // Server should return redirectUrl even on error, this function is a fallback.
        error
        console.error(error)
        await router.replace(`/sign-in?error=${error.message}`)
      },
    }
  )

```

Reply:
done! lmk if I misunderstood what you meant